pants-plugins/src/python/internal_backend/rules_for_testing/register.py

```diff
@@ -9,15 +9,29 @@
 from pants.engine.rules import console_rule
 from pants.engine.selectors import Select
 from pants.rules.core.exceptions import GracefulTerminationException
+from pants.subsystem.subsystem import Subsystem
 
 
-@console_rule('list-and-die-for-testing', [Select(Console), Select(BuildFileAddresses)])
+class ListAndDieForTesting(Subsystem):
+  """A fast and deadly variant of `./pants list`."""
+
+  options_scope = 'list-and-die-for-testing'
+
+
+@console_rule(ListAndDieForTesting, [Select(Console), Select(BuildFileAddresses)])
 def fast_list_and_die_for_testing(console, addresses):
```

Comment:
The correspondence between `options_scope` and goal names now (and having that be the result of an `@console_rule`!!) is an incredible simplification!